### DISCUSSION TITLE
Current Calibration

### DISCUSSION THREAD

**Discussion by jolcese** [emporia-vue-local/esphome#131] [Q&A] [answered]
I have installed a Vue and it's consistently reporting daily energy ~3.5% lower than what I can see on the utility's website.
The voltage readings on both phases are very close to what I can measure with a good Fluke multimeter. 
And the power sum of all the individual circuits at a given time is also around ~3.5% lower than the total power (phase 1 + phase 2)

So my take is that the main current clamps need some calibration. Is there a way to do it with the current code? I only see calibration for voltage.

**Comment by flaviut** [comment, marked as answer]
Every field can have a multiplier or a calibrate_linear filter, and you're
not limited to just integers.

**Reply by flaviut** [reply]
Let me clarify with an example:

Good multimeter reads 2.2A, Vue reads 2A. Either of these two lines can be used to calibrate this out:

```
      - { phase_id: phase_b, input:  "2", power: { name:  "Circuit 2 Power", id:  cir2, filters: [ *moving_avg, *pos, multiply: 1.1 ] } }
```

```
      - { phase_id: phase_b, input:  "2", power: { name:  "Circuit 2 Power", id:  cir2, filters: [ *moving_avg, *pos, calibrate_linear: ["0.0 -> 0.0", "2.0 -> 2.2"] ] } }
```

There's no need to make changes to the daily energy sensors.

**Reply by jolcese** [reply]
Thank you!
